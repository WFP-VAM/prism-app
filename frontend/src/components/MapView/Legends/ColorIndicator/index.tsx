import { createStyles, makeStyles, Typography } from '@material-ui/core';
import { alpha } from '@material-ui/core/styles';

const useStyles = makeStyles(() =>
  createStyles({
    container: {
      display: 'flex',
      alignItems: 'center',
    },
    indicator: {
      height: 10,
      width: 10,
      marginRight: 4,
    },
  }),
);

function ColorIndicator({
  value,
  color,
  opacity,
  fillPattern,
}: ColorIndicatorProps) {
  const classes = useStyles();

  return (
    <div className={classes.container}>
      <div
        className={classes.indicator}
        style={{
          backgroundColor: fillPattern
            ? `linear-gradient(
          to ${fillPattern} bottom, 
          ${color} calc(100% / 7),  
          white calc(100% / 7) calc(100% / 7 * 2), 
          ${color} calc(100% / 7 * 2) calc(100% / 7 * 3), 
          white calc(100% / 7 * 3) calc(100% / 7 * 4), 
          ${color} calc(100% / 7 * 4) calc(100% / 7 * 5),
          white calc(100% / 7 * 5) calc(100% / 7 * 6), 
          ${color} calc(100% / 7 * 6)
        )`
            : alpha(color, opacity),
        }}
      />
      <Typography color="textSecondary">{value}</Typography>
    </div>
  );
}

export interface ColorIndicatorProps {
  value: string;
  color: string;
  opacity: number;
  fillPattern?: 'left' | 'right';
}

export default ColorIndicator;
