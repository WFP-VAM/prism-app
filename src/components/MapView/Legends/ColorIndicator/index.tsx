import React from 'react';
import { createStyles, makeStyles, Typography } from '@material-ui/core';
import { fade } from '@material-ui/core/styles/colorManipulator';

const useStyles = makeStyles(() =>
  createStyles({
    container: {
      display: 'flex',
      alignItems: 'center',
    },
    indicator: {
      height: 10,
      width: 10,
      marginRight: 4,
      backgroundColor: ({
        color,
        opacity,
      }: {
        color: string;
        opacity: number;
      }) => fade(color, opacity),
    },
  }),
);

function ColorIndicator({ value, color, opacity }: ColorIndicatorProps) {
  const classes = useStyles({ color, opacity });
  return (
    <div className={classes.container}>
      <div className={classes.indicator} />
      <Typography color="textSecondary">{value}</Typography>
    </div>
  );
}

export interface ColorIndicatorProps {
  value: string;
  color: string;
  opacity: number;
}

export default ColorIndicator;
