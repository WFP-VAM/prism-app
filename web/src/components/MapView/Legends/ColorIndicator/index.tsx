import React from 'react';
import { createStyles, makeStyles, Typography } from '@material-ui/core';

const useStyles = makeStyles(() =>
  createStyles({
    container: {
      display: 'flex',
      alignItems: 'center',
    },
    indicator: {
      height: 10,
      width: 10,
      marginRight: 4,
      backgroundColor: ({ color }: any) => color,
    },
  }),
);

function ColorIndicator({ value, color }: ColorIndicatorProps) {
  const classes = useStyles({ color });

  return (
    <div className={classes.container}>
      <div className={classes.indicator} />
      <Typography color="textSecondary">{value}</Typography>
    </div>
  );
}

export interface ColorIndicatorProps {
  value: string;
  color: string;
}

export default ColorIndicator;
