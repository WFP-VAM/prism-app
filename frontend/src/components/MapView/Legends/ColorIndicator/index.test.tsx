import React from 'react';
import { render } from '@testing-library/react';

import ColorIndicator from '.';

const props = {
  value: 'Vegetation',
  color: '#EF3202',
  opacity: 0.7,
};

test('renders as expected', () => {
  const { container } = render(<ColorIndicator {...props} />);
  expect(container).toMatchSnapshot();
});
