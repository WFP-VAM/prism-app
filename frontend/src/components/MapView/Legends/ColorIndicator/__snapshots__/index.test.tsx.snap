// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    class="makeStyles-container-1"
  >
    <div
      class="makeStyles-indicator-2"
      style="background-color: rgba(239, 50, 2, 0.7);"
    />
    <mock-typography
      color="textSecondary"
    >
      Vegetation
    </mock-typography>
  </div>
</div>
`;
